src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -220,12 +220,17 @@ public void WriteBlob(BinaryLogRecordKind kind, byte[] bytes)
 
         public void WriteBlob(BinaryLogRecordKind kind, Stream stream)
         {
+            if (stream.Length > int.MaxValue)
+            {
+                throw new ArgumentOutOfRangeException(nameof(stream));
```

Comment:
When we throw this exception, who is the intended recipient? Is it caught anywhere? What are the observable symptoms? If it's not caught, are we ready to crash the process? What is the user experience then?

I don't think we should throw, we should silently skip and resume.

Reply:
Exactly, this branch condition should be `false`, because the caller checks the length. I added it defensively in case some other callers will use this method in future.